itests/eth_api_test.go

```diff
@@ -0,0 +1,35 @@
+package itests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/chain/types/ethtypes"
+	"github.com/filecoin-project/lotus/chain/wallet/key"
+	"github.com/filecoin-project/lotus/itests/kit"
+)
+
+func TestEthAddressToFilecoinAddress(t *testing.T) {
+	client, _, _ := kit.EnsembleMinimal(t, kit.MockProofs(), kit.ThroughRPC())
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	secpKey, err := key.GenerateKey(types.KTDelegated)
```

Comment:
Is this still a secpKey since the type here is delegated?

Reply:
Yes, "Delegated" signatures are solely secp.

Reply:
For my own knowledge,
We also have a secp address type which I presume have secp keys
What is the mapping of address types to key types? Are those orthogonal concepts?

Reply:
So this is a little gnarly. Secp addresses correspond to secp keys. Delegated addresses _today_ solely correspond to secp keys. They will eventually support other kinds of signatures, such as "account abstraction".

So why do we have the KTDelegated sigtype today? It's because the Lotus keystore couples the stored key with its address representation. So when working with keys, we need to treat delegated sigtypes differently from secp sigtypes (eg. [here](https://github.com/filecoin-project/lotus/blob/cdf3812e4081d35692b5ab81c573ab82d0df69cf/chain/wallet/key/key.go#L55-L69)).